firedrake/plot.py

```diff
@@ -1,373 +1,280 @@
-    else:
-        raise NotImplementedError("Plotting functions with geometric dimension %d unsupported",
-                                  gdim)
+    return _plot_2d_field("tricontourf", function, *args, **kwargs)
 
 
-def interactive_multiple_plot(functions, num_sample_points=10, axes=None, **kwargs):
-    """Create an interactive plot for multiple 1D functions to be viewed in
-    Jupyter Notebook
+def tricontour(function, *args, **kwargs):
+    r"""Create a contour plot of a finite element field
 
-    :arg functions: 1D Functions to be plotted
-    :arg num_sample_points: Number of sample points per element, ignore if
-        degree < 4 where Bezier curve is used for an exact plot
-    :arg kwargs: Additional key word arguments to be passed to
-        ``matplotlib.plot``
+    :arg function: the function to plot
+    :return: matplotlib TriContourSet object
     """
-    import matplotlib.pyplot as plt
-    try:
-        from ipywidgets import interact, IntSlider
-    except ImportError:
-        raise RuntimeError("Not in notebook")
+    return _plot_2d_field("tricontour", function, *args, **kwargs)
 
+
+def trisurf(function, *args, **kwargs):
+    r"""Create a 3D surface plot of a finite element field
+
+    :arg function: the function to plot
+    :return:
+    """
+    axes = kwargs.pop("axes", None)
     if axes is None:
-        axes = plt.subplot(111)
+        figure = plt.figure()
+        axes = figure.add_subplot(projection='3d')
 
-    def display_plot(index):
-        axes.clear()
-        return plot(functions[index], num_sample_points, axes=axes, **kwargs).figure
+    if len(function.ufl_shape) == 1:
+        mesh = function.ufl_domain()
+        element = function.ufl_element().sub_elements()[0]
+        Q = FunctionSpace(mesh, element)
+        function = interpolate(sqrt(inner(function, function)), Q)
 
-    interact(display_plot, index=IntSlider(min=0, max=len(functions)-1,
-                                           step=1, value=0),
-             continuous_update=False)
-    return axes
+    num_sample_points = kwargs.pop("num_sample_points", 10)
+    triangulation, vals = _two_dimension_triangle_func_val(function,
+                                                           num_sample_points)
 
+    _kwargs = {"antialiased": False, "edgecolor": "none", "shade": False,
+               "cmap": plt.rcParams["image.cmap"]}
+    _kwargs.update(kwargs)
 
-def piecewise_linear(points, axes=None, **kwargs):
-    """Plot a piece-wise linear plot for the given points, returns a
-    matplotlib axes.
+    return axes.plot_trisurf(triangulation, vals, *args, **_kwargs)
 
-    :arg points: Points to be plotted
-    :arg axes: Axes to be plotted on
-    :arg kwargs: Additional key word arguments passed to plot
+
+def tripcolor(function, *args, **kwargs):
+    r"""Create a pseudo-color plot of a finite element field
+
+    :arg function: the function to plot
+    :return: matplotlib Collection object
```

Comment:
See previous notes about specifying that this plots a Firedrake function (assuming that I've understood capability correctly).

Reply:
Also missing documentation for `args` and `kwargs`